internal/yarpcerrors/yarpcerrors.go

```diff
@@ -31,3 +35,17 @@ import "go.uber.org/yarpc/yarpcerrors"
 func NewWithNamef(code yarpcerrors.Code, name string, format string, args ...interface{}) *yarpcerrors.Status {
 	return yarpcerrors.Newf(code, format, args...).WithName(name)
 }
+
+// AnnotateWithInfo will take an error and add info to it's error message while
+// keeping the same status code.
+func AnnotateWithInfo(err error, format string, args ...interface{}) *yarpcerrors.Status {
```

Comment:
You might want this to be a `*yarpcerrors.Status` instead of an `error` as the argument. This could lead to come confusing outcomes, see below.

Reply:
makes sense, 99% of the cases will likely call with:
```
AnnotateWithInfo(yarpcerrors.FromError(err), ...)
```
But it's probably better to be explicit about that.

Luckily all this is internal API so if we don't like that we can change it later.